src/libsystemd/sd-netlink/netlink-message.c

```diff
@@ -186,6 +190,14 @@ static int add_rtattr(sd_netlink_message *m, unsigned short type, const void *da
         /* get the new message size (with padding at the end) */
         message_length = offset + RTA_ALIGN(rta_length);
 
+        /* Buffer should fit one page as this choice is good for headerless malloc.
+         * Limited to 8K so that userspace to avoid message truncation when
+         * page size is very large. */
```

Comment:
typo: "so that userspace *can* avoid"

The real reason for this, though, is that this is the kernel's default receive buffer size for netlink messages, making it naturally then the good fit for userspace too.